tests/selectors_test.py

```diff
@@ -72,22 +72,121 @@ def test_categorical(
     expected = {"b": ["a", "b", "c"]}
 
     df = nw.from_native(constructor(data)).with_columns(nw.col("b").cast(nw.Categorical))
-    result = df.select(categorical())
+    result = df.select(ncs.categorical())
     assert_equal_data(result, expected)
 
 
+def test_datetime(constructor: Constructor, request: pytest.FixtureRequest) -> None:
+    if (
+        "pyspark" in str(constructor)
+        or "duckdb" in str(constructor)
+        or "dask" in str(constructor)
+        or ("pyarrow_table" in str(constructor) and PYARROW_VERSION < (12,))
+        or ("pyarrow" in str(constructor) and is_windows())
+        or ("pandas" in str(constructor) and PANDAS_VERSION < (2,))
```

Comment:
These are a lot of xfail, but let me go through them:

- pyspark and duckdb: do not implement selectors
- dask: see [comment](https://github.com/narwhals-dev/narwhals/pull/1822#discussion_r1920872606)
- pyarrow < 12, does not implement `replace_time_zone`
- pyarrow in windows: fails to find UTC timezone
- pandas < 2, does not support `time_units!="ns"`

Reply:
> pyspark and duckdb: do not implement selectors

i think we do have these now, right?

Reply:
Yes, but also no!
I can add the selector but we have very minimal support for datetime dtype in spark and duckdb (read as: no support for timezones)